sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -228,14 +232,27 @@ impl InputSelection {
         self
     }
 
+    /// Sets the total mana rewards for required inputs.
+    pub fn with_mana_rewards(mut self, mana_rewards: HashMap<OutputId, u64>) -> Self {
+        self.mana_rewards = mana_rewards;
+        self
+    }
+
+    /// Sets the mana rewards for the given input.
+    pub fn add_mana_rewards(mut self, input: OutputId, mana_rewards: u64) -> Self {
+        self.mana_rewards.insert(input, mana_rewards);
+        self
+    }
+
     fn filter_inputs(&mut self) {
         self.available_inputs.retain(|input| {
             // TODO what about other kinds?
-            // Filter out non basic/account/foundry/nft outputs.
+            // Filter out non basic/account/foundry/nft/delegation outputs.
             if !input.output.is_basic()
                 && !input.output.is_account()
                 && !input.output.is_foundry()
                 && !input.output.is_nft()
+                && !input.output.is_delegation()
```

Comment:
This is annoying, I don't think delegation should be generally available. If we they are not explicitly required/burnt, I don't think they should be available.

Reply:
okay I can remove it